cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -1133,25 +1133,127 @@ public void testNewToPendingProposalDeleteRemoveBroker(VertxTestContext context)
         this.krNewToPendingProposalDelete(context, CruiseControlEndpoints.REMOVE_BROKER, kr);
     }
 
+    private void krNewToPendingProposalDelete(VertxTestContext context, CruiseControlEndpoints endpoint, KafkaRebalance kr) throws IOException, URISyntaxException {
+        // Set up the rebalance endpoint with the number of pending calls before a response is received.
+        cruiseControlServer.setupCCRebalanceResponse(1, endpoint);
+        cruiseControlServer.setupCCUserTasksResponseNoGoals(1, 0);
+
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).create();
+        crdCreateKafka();
+        crdCreateCruiseControlSecrets();
+
+        Checkpoint checkpoint = context.checkpoint();
+        krao.reconcile(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, kr.getMetadata().getName()))
+                // the resource moved from 'New' to 'PendingProposal' (due to the configured Mock server pending calls)
+                .onComplete(context.succeeding(v ->
+                        assertState(context, client, namespace, kr.getMetadata().getName(), KafkaRebalanceState.PendingProposal)))
+                .compose(v -> {
+                    // trigger another reconcile to process the PendingProposal state
+                    return krao.reconcile(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, kr.getMetadata().getName()));
+                })
+                .onComplete(context.succeeding(v ->
+                        assertState(context, client, namespace, kr.getMetadata().getName(), KafkaRebalanceState.PendingProposal)))
+                .compose(v -> {
+                    // trigger another reconcile to process the PendingProposal state
+                    Crds.kafkaRebalanceOperation(client).inNamespace(namespace).withName(kr.getMetadata().getName()).delete();
+                    return krao.reconcile(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, kr.getMetadata().getName()));
+                })
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    // the resource should not exist anymore
+                    KafkaRebalance currentKR = Crds.kafkaRebalanceOperation(client).inNamespace(namespace).withName(kr.getMetadata().getName()).get();
+                    assertThat(currentKR, is(nullValue()));
+                    checkpoint.flag();
+                })));
+    }
+
     /**
      * Tests the transition from 'Rebalancing' to 'NotReady' due to a user task not existing and an error
-     * on re-issuing the optimization proposal request
+     * on re-issuing the optimization proposal request (i.e. brokers don't exist during a "remove-broker")
```

Comment:
```suggestion
     * on re-issuing the optimization proposal request (i.e. brokers don't exist during a "remove-broker")
```
* on re-issuing the optimization proposal request (i.e. brokers don't exist during the "remove-broker" operation)